### PR TITLE
Refactor super() constructors to use Dart 2.17 features. Completed. Please review.

### PR DESCRIPTION
**Pull request by IgorBabalich**

### CODE REVIEW

File: ngcompiler/lib/v1/src/compiler/attribute_matcher.dart [hunk 1]
```diff
@@ -9,7 +9,7 @@ abstract class AttributeMatcher {
   final String name;
   final String? value;
 
-  const AttributeMatcher(this.name, this.value);
+  const AttributeMatcher(this.name, [this.value]);
```
Comment (GZGavinZhao) [inline ngcompiler/lib/v1/src/compiler/attribute_matcher.dart:12]: Why is this necessary?

**Review by GZGavinZhao — CHANGES_REQUESTED** [review]
Great work! Just one little thing. Please see the comment.

File: ngcompiler/lib/v1/src/compiler/attribute_matcher.dart (hunk 1 above)
Comment (IgorBabalich) [reply ngcompiler/lib/v1/src/compiler/attribute_matcher.dart:12]: In order to resolve ugly null parameter passing to super() below on line 81, by making second nullable parameter as optional:
class SetAttributeMatcher extends AttributeMatcher {  
--     SetAttributeMatcher(String name) : super(name, null);
++   SetAttributeMatcher(super.name);

**Comment by GZGavinZhao** [comment]
Squashed and merged. The pull request will say it's not merged, but it's just that GitHub is not smart enough to figure that out.
